app/jobs/autoscaling.py

```diff
@@ -23,14 +25,22 @@
 SCALING_METRIC_TAG_PREFIX = "IDSeqScalingData_"
 SCALING_METRIC_TAG_FORMAT = SCALING_METRIC_TAG_PREFIX + "{environment}"
 
-
 # A value recorded this many minutes earlier will be ignored
 EXPIRATION_PERIOD_MINUTES = 60
 
-
 DEBUG = False
 
 
+## constants specific to draining / safe scale-down:
+DRAINING_TAG = "draining"
+MAX_REFRESH_INTERVAL = 900
+# MAX_REFRESH_INTERVAL must be the same as MAX_INTERVAL_BETWEEN_DESCRIBE_INSTANCES in idseq-dag.
+# After this interval, we are sure we are safe from race conditions related to batch jobs dispatching gsnap/rapsearch jobs.
+JOB_TAG_PREFIX = "RunningIDseqBatchJob_"
+ALIGNMENT_JOB_EXPIRATION_SECONDS = 120 * 60
```

Comment:
2 hours?

Reply:
Pretty arbitrary, what time period would you recommend based on your benchmarking?

Reply:
Only if there are crashed jobs on a server. Normally job tags are removed from servers when job succeeds. To handle the case where a crash prevented the tag from being removed, we clean up old tags every ALIGNMENT_JOB_EXPIRATION_SECONDS seconds.

So ALIGNMENT_JOB_EXPIRATION_SECONDS does need to be long enough that we are certain the job tag should be discarded. However, I agree, we should make it as short as possible while fulfilling this constraint. Will ask @boris-dimitrov what the distribution of job runtimes is.

Reply:
I guess what I meant here was more along the lines of changing the meaning of that constant, not the behavior of this code.

Reply:
ok. I'm changing ALIGNMENT_JOB_EXPIRATION_SECONDS to 30 * 60 and will add the alive signal in idseq-dag